packages/amplify-cli/src/commands/helpers/projectUtils.ts

```diff
@@ -1,11 +1,15 @@
-import { pathManager } from 'amplify-cli-core';
+import { AmplifyError, pathManager } from 'amplify-cli-core';
 
 /**
  * checks if the current directory is a nested project
  */
 export const checkForNestedProject = (): void => {
   const projectRoot = pathManager.findProjectRoot() ?? process.cwd();
   if (projectRoot !== process.cwd()) {
-    throw new Error(`Creating a nested amplify project is not supported. Project root detected: ${projectRoot}`);
+    throw new AmplifyError('ProjectInitError', {
```

Comment:
I'm not sure how fine-grained we want to get with the error types, but perhaps it makes sense for this one to be `NestedInitError`